airflow/contrib/operators/sagemaker_transform_operator.py

```diff
@@ -0,0 +1,123 @@
+        If given, it will be used to create a SageMaker model before creating
+        the SageMaker transform job
+    :type model_config: dict
+    :param aws_conn_id: The AWS connection ID to use.
+    :type aws_conn_id: string
+    :param wait_for_completion: if the program should keep running until job finishes
+    :type wait_for_completion: bool
+    :param check_interval: if wait is set to be true, this is the time interval
+        in seconds which the operator will check the status of the transform job
+    :type check_interval: int
+    :param max_ingestion_time: if wait is set to be true, the operator will fail
+        if the transform job hasn't finish within the max_ingestion_time in seconds
+        (Caution: be careful to set this parameters because transform can take very long)
+    :type max_ingestion_time: int
+    """
+
+    @apply_defaults
+    def __init__(self,
+                 config,
+                 aws_conn_id='aws_default',
+                 wait_for_completion=True,
+                 check_interval=30,
+                 max_ingestion_time=None,
+                 *args, **kwargs):
+        super(SageMakerTransformOperator, self).__init__(config=config,
+                                                         aws_conn_id=aws_conn_id,
+                                                         *args, **kwargs)
+
+        self.aws_conn_id = aws_conn_id
+        self.config = config
+        self.wait_for_completion = wait_for_completion
+        self.check_interval = check_interval
+        self.max_ingestion_time = max_ingestion_time
+        self.create_integer_fields()
+
+    def create_integer_fields(self):
+        self.integer_fields = [
+            ['Transform', 'TransformResources', 'InstanceCount'],
+            ['Transform', 'MaxConcurrentTransforms'],
+            ['Transform', 'MaxPayloadInMB']
+        ]
+        if 'Transform' not in self.config:
+            for field in self.integer_fields:
+                field.pop(0)
+
+    def expand_role(self):
+        if 'Model' not in self.config:
+            return
+        config = self.config['Model']
+        if 'ExecutionRoleArn' in config:
+            config['ExecutionRoleArn'] = \
+                self.hook.expand_role(config['ExecutionRoleArn'])
+
+    def execute(self, context):
+        self.preprocess_config()
+
+        model_config = self.config['Model']\
+            if 'Model' in self.config else None
+        transform_config = self.config['Transform']\
+            if 'Transform' in self.config else self.config
+
+        if model_config:
+            self.log.info(
+                'Creating SageMaker Model %s for transform job'
+                % model_config['ModelName']
+            )
+            self.hook.create_model(model_config)
+
+        self.log.info(
+            'Creating SageMaker transform Job %s.'
+            % transform_config['TransformJobName']
+        )
```

Comment:
self.config can be in two format
Either a combination of both model and transform config
{
'Model': { ModelConfig }
'Transform': { TransformConfig }
}

Or just transform config(same as the one in above after 'Transform')
{ TransformConfig }

So we will evaluate transform_config. It's either self.config or self.config['Transform']. Hence when we reach this line, transform_config will never have a key 'Transform' here.

Reply:
Ah, I didn't get that point from the docs. (I might have missed it/not gotten to that point yet).

Reply:
Yep you are right. The doc is not clear about that. But is that good if I fix the doc in next PR? I need all docstrings to be reviewed by tech writer first. To make things faster, I guess making it a requirement for next PR instead of blocking this PR would be a better idea.

Reply:
This is _probably_ okay, yeah.